app/services/watermark_service.rb

```diff
@@ -0,0 +1,121 @@
+class WatermarkService
+  POINTSIZE = 20
+  KERNING = 1.2
+  ANGLE = 45
+  FILL_COLOR = "rgba(0,0,0,0.4)"
+
+  attr_reader :text
+  attr_reader :text_length
+
+  def initialize(text = APPLICATION_NAME)
+    @text = " #{text} " # give more space around each occurence
+    @text_length = @text.length
+  end
+
+  def process(filepath, output = nil)
+    file = Rails.root.join(filepath)
+    metadata = image_metadata(file)
```

Comment:
quand tu fais un blob.open, tu as un tempfile, du coup, tu pourrais faire directement un 
```suggestion
    metadata = image_metadata(file)
```